CONTRIBUTION_LADDER.md

```diff
@@ -0,0 +1,105 @@
+# Contribution Ladder
```

Comment:
I would rather link people to a common ladder in Porter's main repo so that we aren't maintaining separate copies of this content, which is getting stale and will need to be updated soon. Let's figure out where this is linked to from our repo, and just link directly to the core porter repo ladder.